heka/sandbox/decoders/extract_telemetry_dimensions.lua

```diff
@@ -12,13 +12,29 @@ local msg = {
 Timestamp   = nil,
 Type        = "telemetry",
 Payload     = nil,
-Fields      = { sourceName = "telemetry" }
+EnvVersion  = 1
 }
 
 local UNK_DIM = "UNKNOWN"
 local UNK_GEO = "??"
+local environment_objects = {"build", "settings", "profile", "partner", "system", "addons"}
+local main_ping_objects = {"info", "simpleMeasurements", "histograms", "keyedHistograms", "chromeHangs", "threadHangStats", "log", "fileIOReports", "lateWrites", "addonDetails", "addonHistograms", "UIMeasurements", "slowSQL", "slowSQLstartup", "childPayloads"}
+
+local function split_objects(root, section, objects)
+    if type(root) ~= "table" then return end
+
+    for i, name in ipairs(objects) do
+        if type(root[name]) == "table" then
+            local ok, json = pcall(cjson.encode, root[name])
+            if ok then
+                msg.Fields[string.format("%s.%s", section, name)] = json
```

Comment:
We are in full control of the message, they don't exist.  Also, even if they did I believe the correct behaviour would be to overwrite otherwise they could be hijacked (unless you want to fail the message?).